dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift

```diff
@@ -78,69 +78,78 @@ class dydxMarketPositionViewPresenter: HostedViewPresenter<dydxMarketPositionVie
-            let stepSizeDecimals = market.configs?.stepSizeDecimals?.intValue ?? 0
-            if takeProfitOrders.count > 1 {
-                viewModel?.takeProfitStatusViewModel = .init(
-                    triggerSide: .takeProfit,
-                    triggerPriceText: DataLocalizer.shared?.localize(path: "APP.TRADE.MULTIPLE_ARROW", params: nil),
-                    action: routeToOrdersAction)
-            } else if let takeProfitOrder = takeProfitOrders.first {
-                viewModel?.takeProfitStatusViewModel = .init(
-                    triggerSide: .takeProfit,
-                    triggerPriceText: dydxFormatter.shared.raw(number: takeProfitOrder.triggerPrice?.doubleValue, digits: decimalDigits),
-                    limitPrice: takeProfitOrder.type == .takeprofitlimit ? dydxFormatter.shared.raw(number: takeProfitOrder.price, digits: decimalDigits) : nil,
-                    amount: position.size?.current?.doubleValue.magnitude == takeProfitOrder.size.magnitude ? nil : dydxFormatter.shared.raw(number: takeProfitOrder.size, digits: stepSizeDecimals),
-                    action: routeToTakeProfitStopLossAction)
-            } else {
-                viewModel?.takeProfitStatusViewModel = .init(
-                    triggerSide: .takeProfit,
-                    action: routeToTakeProfitStopLossAction)
+            let routeToOrdersAction = {
+                Router.shared?.navigate(to: RoutingRequest(path: "/market", params: ["currentSection": "orders"]), animated: true, completion: nil)
+                return
             }
 
-            if stopLossOrders.count > 1 {
-                viewModel?.stopLossStatusViewModel = .init(
-                    triggerSide: .stopLoss,
-                    triggerPriceText: DataLocalizer.shared?.localize(path: "APP.TRADE.MULTIPLE_ARROW", params: nil),
-                    action: routeToOrdersAction)
-            } else if let stopLossOrder = stopLossOrders.first {
-                viewModel?.stopLossStatusViewModel = .init(
-                    triggerSide: .stopLoss,
-                    triggerPriceText: dydxFormatter.shared.raw(number: stopLossOrder.triggerPrice?.doubleValue, digits: decimalDigits),
-                    limitPrice: stopLossOrder.type == .stoplimit ? dydxFormatter.shared.raw(number: stopLossOrder.price, digits: decimalDigits) : nil,
-                    // don't show amount unless order size is custom
-                    amount: position.size?.current?.doubleValue.magnitude == stopLossOrder.size.magnitude ? nil : dydxFormatter.shared.raw(number: stopLossOrder.size, digits: stepSizeDecimals),
-                    action: routeToTakeProfitStopLossAction)
+            let takeProfitOrders = triggerOrders.filter { (order: SubaccountOrder) in
+                order.marketId == position.id
+                && (order.type == .takeprofitmarket || (order.type == .takeprofitlimit && AbacusStateManager.shared.environment?.featureFlags.isSlTpLimitOrdersEnabled == true))
```

Comment:
added this ff check

Reply:
Yeah.. I need to add it to Android as well